src/AutoMapper/MapMemberInfo.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.ComponentModel;
+using System.Reflection;
+
+namespace AutoMapper
+{
+    /// <summary>
+    /// Used to ensure we capture the concrete parent type. For virtual members, AutoMapper's GetMemberInfo methods will return a
+    /// MemberInfo whose ReflectedType is the base class - not source or destination being mapped.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class MapMemberInfo : MemberInfo, IEquatable<MemberInfo>
+    {
+        public MapMemberInfo(MemberInfo memberInfo, Type parentType)
+        {
+            _memberInfo = memberInfo;
+            _parentType = parentType;
+        }
+
+        private MemberInfo _memberInfo { get; set; }
+        private Type _parentType { get; set; }
+
+        public override Type DeclaringType => _memberInfo.DeclaringType;
+
+        public override MemberTypes MemberType => _memberInfo.MemberType;
+
+        public override string Name => _memberInfo.Name;
+
+        public override Type ReflectedType => _parentType;
+
+        public bool Equals(MemberInfo other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+
+            return this._memberInfo.Equals(other);
```

Comment:
Always `false`. My co-workers traced a memory leak back to this line. The projection cache keeps growing in 10.1.1. No longer the case now, so it's fixed in 11.0.

Reply:
Yep -  here's from PropertyInfo source:
![image](https://user-images.githubusercontent.com/7595209/123402859-2f382d00-d576-11eb-84dd-5e5532a0437f.png)

MemberInfo does the same.  Thanks for the heads-up.